tests/devapps/Intune-xamarin-Android/MainActivity.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using Android.App;
+using Android.OS;
+using Android.Runtime;
+using AndroidX.AppCompat.Widget;
+using AndroidX.AppCompat.App;
+using Microsoft.Identity.Client;
+using Android.Content;
+using Microsoft.Intune.Mam.Policy;
+using Microsoft.Intune.Mam.Client.App;
+using System.Threading.Tasks;
+
+namespace Intune_xamarin_Android
+{
+    [Activity(Label = "@string/app_name", Theme = "@style/AppTheme.NoActionBar", MainLauncher = true)]
+    public class MainActivity : AppCompatActivity
+    {
+        /// <summary>
+        /// The scopes that are protected by conditional access
+        /// </summary>
+        internal static string[] Scopes = { "api://09aec9b9-0b0f-488a-81d6-72fd13a3a1c1/Hello.World" };
+
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+            Xamarin.Essentials.Platform.Init(this, savedInstanceState);
+            SetContentView(Resource.Layout.activity_main);
+
+            Toolbar toolbar = FindViewById<Toolbar>(Resource.Id.toolbar);
+            SetSupportActionBar(toolbar);
+            
+            Android.Widget.Button actButton = FindViewById<Android.Widget.Button>(Resource.Id.acqToken);
+            actButton.Click += ActButton_Click;
+
+            Android.Widget.Button signOutButton = FindViewById<Android.Widget.Button>(Resource.Id.signOut);
+            signOutButton.Click += SignOutButton_Click;
+        }
+
+        /// <summary>
+        /// This method shows calling pattern to access resource protected by Conditional Access with App Protection Policy
+        /// </summary>
+        /// <param name="sender">Sender button</param>
+        /// <param name="e">arguments</param>
+        private async void ActButton_Click(object sender, EventArgs e)
+        {
+            AuthenticationResult result = null;
+
+            try
+            {
+                // attempt silent login.
+                // If this is very first time and the device is not enrolled, it will throw MsalUiRequiredException
+                // If the device is enrolled, this will succeed.

```

Comment:
Nit: It will still throw, since no token in cache?

Reply:
Removed the redundant chk.